haystack/preview/components/file_converters/azure.py

```diff
@@ -22,22 +23,33 @@ class AzureOCRDocumentConverter:
     to set up your resource.
     """
 
-    def __init__(self, endpoint: str, api_key: str, model_id: str = "prebuilt-read"):
+    def __init__(self, endpoint: str, api_key: Optional[str] = None, model_id: str = "prebuilt-read"):
         """
         Create an AzureOCRDocumentConverter component.
 
         :param endpoint: The endpoint of your Azure resource.
-        :param api_key: The key of your Azure resource.
+        :param api_key: The key of your Azure resource. It can be
+        explicitly provided or automatically read from the
+        environment variable CORE_AZURE_CS_API_KEY (recommended).
         :param model_id: The model ID of the model you want to use. Please refer to [Azure documentation](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/choose-model-feature)
             for a list of available models. Default: `"prebuilt-read"`.
         """
         azure_import.check()
 
+        if api_key is None:
+            try:
+                api_key = os.environ["CORE_AZURE_CS_API_KEY"]
```

Comment:
I know that internally (in our tests), we set up an environment variable with this name (`CORE_AZURE_CS_API_KEY`). 

However, this name does not seem very clear to users.
I suggest changing it to `AZURE_AI_API_KEY`.

Reply:
Thanks! I will rename the environment variable to `AZURE_AI_API_KEY`.